### PR TITLE
feat: add old parent in tree diff

### PR DESCRIPTION
**Pull request by Leeeon233**

### CODE REVIEW

File: loro-js/tests/tree.test.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-import { LoroDoc, LoroTree, LoroTreeNode } from "../src";
+import { LoroDoc, LoroTree, LoroTreeNode, TreeDiff } from "../src";
 
 function assertEquals(a: any, b: any) {
   expect(a).toStrictEqual(b);
 }
 
 describe("loro tree", () => {
   const loro = new LoroDoc();
+  loro.setWithFractionalIndex(true);
```
Comment (zxch3n) [inline loro-js/tests/tree.test.ts:10]: What's the default value for this?

**Review by zxch3n — CHANGES_REQUESTED** [review]
Tests failed

File: loro-js/tests/tree.test.ts (hunk 1 above)
Comment (Leeeon233) [reply loro-js/tests/tree.test.ts:10]: It's `false`, which means each fractional index will be `Default::default`.

File: crates/loro-common/src/error.rs [hunk 2]
```diff
@@ -88,6 +88,8 @@ pub enum LoroTreeError {
     TreeNodeNotExist(TreeID),
     #[error("The index({index}) should be <= the length of children ({len})")]
     IndexOutOfBound { len: usize, index: usize },
+    #[error("Fractional index is not enabled, you should enable it on the doc first.")]
```
Comment (zxch3n) [inline crates/loro-common/src/error.rs:91]: enable it first by xxx

**Review by zxch3n — APPROVED** [review]

**Pull request merged**